tests/test_date.py

```diff
@@ -132,76 +204,171 @@ def test_periods(self, low, high, period, expected_results):
+        self.then_current_year_for_dates_without_year_is_not_none()
+        self.then_period_for_current_year_is_equal()
+        self.then_check_if_dates_are_equal()
+        self.given_date(datetime(2014, 2, 28).date())
+        self.given_twelfth_second()
+        self.given_datetime_mock_second()
+        self.when_patch_second()
+        self.then_current_year_for_dates_without_year_is_not_none()
+        self.then_period_for_current_year_is_equal()
+        self.then_check_if_dates_are_equal()
+
+    def given_current_year(self):
+        self.current_year = datetime.today().year
+
+    def given_date(self, date_obj):
+        self.date_obj = date_obj
+
+    def when_should_use_current_year_for_dates_without_year(self):
+        self.result = date.parse_with_formats('09.16', ["%m.%d"])
+
+    def then_should_use_current_year_for_dates_without_year(self):
+        self.assertEquals(datetime(self.current_year, 9, 16).date(), self.result['date_obj'].date())
+
+    def when_should_parse_date(self):
+        self.result = date.parse_with_formats('25-03-14', ['%d-%m-%y'])
+
+    def then_should_parse_date(self):
+        self.assertEquals(datetime(2014, 3, 25).date(), self.result['date_obj'].date())
+
+    def then_shouldnt_parse_invalid_date(self):
+        self.assertIsNone(date.parse_with_formats('yesterday', ['%Y-%m-%d'])['date_obj'])
+
+    def given_twelfth_first(self):
+        self.twelfth_first = datetime(2014, 8, 12)
 
-        with patch('dateparser.date_parser.datetime', new=datetime_mock):
-            dt_data = date.parse_with_formats('August 2014', ['%B %Y'])
+    def given_twelfth_second(self):
+        self.twelfth_second = datetime(2014, 2, 12)
 
-        self.assertIsNotNone(dt_data)
-        self.assertEquals('month', dt_data['period'])
-        self.assertEquals(datetime(2014, 8, 31).date(), dt_data['date_obj'].date())
+    def given_datetime_mock_first(self):
+        self.datetime_mock = Mock(wraps=datetime)
+        self.datetime_mock.utcnow = Mock(return_value=self.twelfth_first)
 
-        twelfth = datetime(2014, 2, 12)
-        datetime_mock = Mock(wraps=datetime)
-        datetime_mock.utcnow = Mock(return_value=twelfth)
+    def given_datetime_mock_second(self):
+        self.datetime_mock = Mock(wraps=datetime)
+        self.datetime_mock.utcnow = Mock(return_value=self.twelfth_second)
 
-        with patch('dateparser.date_parser.datetime', new=datetime_mock):
-            dt_data = date.parse_with_formats('February 2014', ['%B %Y'])
+    def when_patch_first(self):
+        with patch('dateparser.date_parser.datetime', new=self.datetime_mock):
+            self.dt_data = date.parse_with_formats('August 2014', ['%B %Y'])
 
-        self.assertIsNotNone(dt_data)
-        self.assertEquals('month', dt_data['period'])
-        self.assertEquals(datetime(2014, 2, 28).date(), dt_data['date_obj'].date())
+    def when_patch_second(self):
+        with patch('dateparser.date_parser.datetime', new=self.datetime_mock):
```

Comment:
Let's be consistent all the way and use `add_patch` method in `given_*` instead.